File: erlang/rna-transcription/src/rna_transcription.erl
-module(rna_transcription).

-export([to_rna/1]).

-define(Nucleotides, #{$G => $C, $C => $G, $T => $A, $A => $U}).


to_rna(_Strand) -> [maps:get(Nt, ?Nucleotides) || Nt <- _Strand].

File: fortran/rna-transcription/rna_transcription.f90
module rna_transcription
  implicit none
contains
  function to_rna(dna)
      character(*) :: dna
      character(len(dna)) :: to_rna
      integer :: i
      to_rna = ""
      do i = 1, len(dna(:))
        if (dna(i:i) == "G") then
          to_rna(i:i) = "C"
        else if (dna(i:i) == "C") then
          to_rna(i:i) = "G"
        else if (dna(i:i) == "T") then
          to_rna(i:i) = "A"
        else if (dna(i:i) == "A") then
          to_rna(i:i) = "U"
        end if
      end do
    return
  end function to_rna
end module rna_transcription

